File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include <queue>
#include <cmath>
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int ret = 0;
        int level = 1;
        int curr_max = -pow(10,5);
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            int tot  =0;
            int n = q.size();
            for(int i = 0; i<n; i++){
                TreeNode* curr = q.front();
                q.pop();
                if (curr != nullptr){
                    tot +=curr->val;
                    if(curr->left != nullptr){
                        q.push(curr->left);
                    }
                    if(curr->right != nullptr){
                        q.push(curr->right);
                    }
                }
            }
            if (curr_max<tot){
                ret = level;
                curr_max = tot;
            }
            level++;

        }

        return ret;
    }
};

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        level_sum = {}
        queue = [(root,1)]
        while queue:
            curr,level = queue.pop(0)
            if level in level_sum:
                level_sum[level] +=curr.val
            else:
                level_sum[level] =curr.val
            if curr.left:
                queue.append((curr.left,level+1))
            if curr.right:
                queue.append((curr.right,level+1))
        ret,curr_max = 0,float("-inf")
        keys = list(level_sum.keys())
        for k in keys:
            if curr_max<level_sum[k]:
                ret = k
                curr_max = level_sum[k]
        return ret
